compiler/rustc_middle/src/ty/fold.rs

```diff
@@ -876,6 +876,52 @@ impl<'tcx> PredicateVisitor<'tcx> for HasEscapingVarsVisitor {
     }
 }
 
+crate struct CountBoundVars {
+    crate outer_index: ty::DebruijnIndex,
+    crate bound_tys: FxHashSet<ty::BoundTy>,
+    crate bound_regions: FxHashSet<ty::BoundRegion>,
+    crate bound_consts: FxHashSet<ty::BoundVar>,
+}
```

Comment:
it probably makes sense to use `MiniSet` instead of `FxHashSet` here as these sets should mostly be nearly empty